python/ray/serve/_private/replica.py

```diff
@@ -793,10 +799,21 @@ async def call_user_method_generator(
                 not request_metadata.is_http_request
             ), "HTTP requests should go through `call_user_method`."
             user_method = self.get_runner_method(request_metadata)
-            method_to_call = wrap_generator_function_in_async_if_needed(user_method)
-
-            async for result in method_to_call(*request_args, **request_kwargs):
-                yield result
+            result_generator = user_method(*request_args, **request_kwargs)
+            if inspect.iscoroutine(result_generator):
```

Comment:
Could we also support this pattern:

```python
class C:

    def inner(self):
        for i in range(5):
            yield i

    def outer(self):
        return self.inner()
```

`outer` isn't a coroutine here (it's not async), but it still returns a generator.

Reply:
this works

Reply:
Ah, just saw the `SyncStreamer` test. Cool, I'll resolve this.